qt/FreeSpaceLabel.cc

```diff
@@ -73,33 +66,35 @@ FreeSpaceLabel::onTimer ()
   tr_variantInitDict (&args, 1);
   tr_variantDictAddStr (&args, TR_KEY_path, myPath.toUtf8 ().constData());
 
-  myTag = mySession->getUniqueTag ();
-  mySession->exec ("free-space", &args, myTag);
-}
-
-void
-FreeSpaceLabel::onSessionExecuted (int64_t tag, const QString& result, tr_variant * arguments)
-{
-  Q_UNUSED (result);
+  RpcQueue * q = new RpcQueue;
 
-  if (tag != myTag)
-    return;
-
-  QString str;
-
-  // update the label
-  int64_t bytes = -1;
-  if (tr_variantDictFindInt (arguments, TR_KEY_size_bytes, &bytes) && bytes >= 0)
-    setText (tr("%1 free").arg(Formatter::sizeToString (bytes)));
-  else
-    setText (QString ());
-
-  // update the tooltip
-  size_t len = 0;
-  const char * path = 0;
-  tr_variantDictFindStr (arguments, TR_KEY_path, &path, &len);
-  str = QString::fromUtf8 (path, len);
-  setToolTip (QDir::toNativeSeparators (str));
+  q->addReq (
+    [this, &args] (RpcResponse)
```

Comment:
In my eyes this is not enough to justify passing it by value down the road. Again, *conceptually* response object should not be modified or copied or moved inside the callback. Copying still means allocation + counter operations + copying itself, which could all be eliminated. Granted, this code is not currently performance-critical, but we should still strive to make it as efficient as it could be, especially when it's simply the right thing to do.

And as I said, this is not the only place. Object of classes such as `QString`, `QFuture<T>`, `QFutureInterface<T>` shouldn't be passed by value as well. Take Qt itself as a reference: it may return those objects by value (which may sometimes be the only acceptable way), but general rule is that all the functions accept non-POD input parameters by const reference.